packages/plugin-truffle/src/utils/deploy-impl.ts

```diff
@@ -40,7 +41,16 @@ export async function deployImpl(
   }
 
   return await fetchOrDeploy(version, provider, async () => {
-    const deployment = await deploy(requiredOpts.deployer, Contract);
+    const deployment = await deploy(requiredOpts.deployer, Contract, ...requiredOpts.constructorArgs);
     return { ...deployment, layout };
   });
 }
+
+function encodeArgs(Contract: ContractClass, constructorArgs: unknown[]): string {
+  const fragment = (Contract as any).abi.find((entry: any) => entry.type == 'constructor');
+
+  return (Contract as any).web3.eth.abi.encodeParameters(
```

Comment:
There should be a global `web3` instance. Do you know if this works?

```suggestion
  return web3.eth.abi.encodeParameters(
```

Reply:
Cannot find name 'web3'